src/common/commonutils/CommandUtils.c

```diff
@@ -15,6 +15,44 @@ static long MonotonicTime()
     return -1;
 }
 
+struct mockCommand {
+    const char* expectedCommand;
+    bool matchPrefix;
+    const char* output;
+    int returnCode;
+    struct mockCommand* next;
+};
+
+static struct mockCommand* g_mockCommand;
+
+void AddMockCommand(const char* expectedCommand, bool matchPrefix, const char* output, int returnCode)
+{
+    struct mockCommand* mock;
+    mock = malloc(sizeof(struct mockCommand));
+    // Handling memory error in a brutal way as it's only for tests, not for production usage.
```

Comment:
> // Handling memory error in a brutal way as it's only for tests, not for production usage. [](http://example.com/codeflow?start=3&length=91)

Doesn't this warrant to have this mocking code as part of the unit-tests instead?

Reply:
Maybe a good opportunity to add a new common test library, that all tests can link to and use?

Reply:
Unfortunately gtest/gmock can't mock "free" (non-C++-class-member) functions, so this code needs to be close to ExecuteCommand itself. We could move AddMock* to a separate unit, but then we'd have to export g_mockCommand, and the "execution" part would still be left in ExecuteCommand body - so I don't think it's worth it.